src/mjbots_simulation_interface.cpp

```diff
@@ -0,0 +1,187 @@
+                                                 int imu_rate_hz,
+                                                 std::shared_ptr<::kodlab::IMUData<float>> imu_data_ptr,
+                                                 std::optional<::mjbots::pi3hat::Euler> imu_world_offset_deg,
+                                                 bool dry_run,
+                                                 bool print_torques,
+                                                 bool send_pd_commands)
+    : imu_data_(imu_data_ptr ? imu_data_ptr : std::make_shared<::kodlab::IMUData<float>>()),
+      dry_run_(dry_run),
+      print_torques_(print_torques),
+      send_pd_commands_(send_pd_commands)
+{ 
+  LOG_IF_WARN(dry_run_, "\nDRY RUN: NO TORQUES COMMANDED");
+  joints = joint_ptrs;
+  num_joints_ = joints.size();
+
+  
+
+  if(imu_world_offset_deg.has_value()){
+    kodlab::rotations::EulerAngles<float> imu_world_offset = {
+        static_cast<float>(M_PI / 180.0 * imu_world_offset_deg->roll),
+        static_cast<float>(M_PI / 180.0 * imu_world_offset_deg->pitch),
+        static_cast<float>(M_PI / 180.0 * imu_world_offset_deg->yaw)
+        };
+    imu_data_->set_world_offset(imu_world_offset.ToQuaternion());
+  }
+
+  // Initialize and send basic command
+  // InitializeCommand();
+}
+
+
+void MjbotsSimulationInterface::Init() {
+    // activate software
+    mj_activate("~/.mujoco/mujoco-2.3.3/bin/mjkey.txt");
+
+    // load and compile model
+    char error[1000] = "Could not load binary model";
+    m = mj_loadXML(xml_model_path.c_str(), 0, error, 1000);
+    if( !m )
+      mju_error_s("Load model error: %s", error);
+
+    // set timestep
+    m->opt.timestep=1.0/control_frequency;
+    // make data
+    d = mj_makeData(m);
+
+
+    // init GLFW
+    if( !glfwInit() )
+        mju_error("Could not initialize GLFW");
+
+    // create window, make OpenGL context current, request v-sync
+    window = glfwCreateWindow(1244, 700, "Demo", NULL, NULL);
+    glfwMakeContextCurrent(window);
+    glfwSwapInterval(1);
+
+    // initialize visualization data structures
+    mjv_defaultCamera(&cam);
+    mjv_defaultOption(&opt);
+    mjv_defaultScene(&scn);
+    mjr_defaultContext(&con);
+    mjv_makeScene(m, &scn, 2000);                // space for 2000 objects
+    mjr_makeContext(m, &con, mjFONTSCALE_150);   // model-specific context
+
+    // install GLFW mouse and keyboard callbacks
+    // glfwSetKeyCallback(window, [](GLFWwindow* window_, int key, int scancode, int action, int mode){ MjbotsSimulationInterface::keyboard(window_, key, scancode, action, mode) ;});
+    glfwSetWindowUserPointer(window, this);
+    glfwSetKeyCallback(window,keyCallbackStatic);
+    glfwSetCursorPosCallback(window, mouseMoveCallbackStatic);
+    glfwSetMouseButtonCallback(window, mouseCallbackStatic);
+    glfwSetScrollCallback(window, scrollCallbackStatic);
+    
+    // initial position
+    d->qpos[0] = 1.57;
```

Comment:
ZF says this should be removed

Reply:
Add initial_positions and initial_velocities in options and use them for MuJoCo simulation